Objective-C/CBLReplicator.mm

```diff
@@ -261,6 +264,26 @@ - (bool) _setupC4Replicator: (C4Error*)outErr {
         .socketFactory = &socketFactory,
     };
     
+    NSUInteger collectionCount = _config.collectionConfigs.count;
+    C4ReplicationCollection cols[collectionCount];
+    NSUInteger i = 0;
+    for (CBLCollection* col in _config.collectionConfigs) {
+        CBLCollectionConfiguration* colConfig = _config.collectionConfigs[col];
+        AllocedDict dict = [self encodedCollectionOptions: colConfig];
+        C4ReplicationCollection c = {
+            .collection = col.c4spec,
+            .push = mkmode(isPush(_config.replicatorType), _config.continuous),
+            .pull = mkmode(isPull(_config.replicatorType), _config.continuous),
```

Comment:
The same .push and .pull setup here should be removed as the default collection when using database should be in the collectionConfigs as well:

https://github.com/couchbase/couchbase-lite-ios/pull/2987/files#diff-db9ac2c53f3d61c425b37311410aa8be9d6fcafb43e203cd6235c63b18455b0bR250-R251

Reply:
but the 'CBLCollectionConfiguration' class doesn't include the ReplicatorMode(push/pull) in it, and we are not copying the values as well.

Reply:
ok. I think I get it, I will remove the C4ReplicatorParameters.push/pull setting, as the default collection config also is present here.